static/app/stores/organizationsStore.tsx

```diff
@@ -36,7 +36,19 @@ const storeConfig: OrganizationsStoreDefinition = {
   },
 
   onUpdate(org) {
-    this.add(org);
+    let match = false;
+    this.state.forEach((existing, idx) => {
+      if (existing.id === org.id) {
+        this.state[idx] = {...existing, ...org};
+        match = true;
+      }
+    });
+    if (!match) {
+      throw new Error(
+        'Cannot update an organization that is not in the OrganizationsStore'
+      );
```

Comment:
Could this be hit by `fetchOrg()` in actionCreators/organization. It isn't clear that the organization will already be in the client when the organization is fetched again.

Reply:
Yeah let me take a closer look, that's kind of what I was worried about.

I think in that case it can just call `add` tho

Reply:
Actually I don't see any usage of this that looks dangerous. I think we're okay.